src/v2/Apps/Auction/Components/BidForm.tsx

```diff
@@ -235,6 +236,9 @@ export const BidForm: React.FC<Props> = ({
 
                     <CreditCardInput
                       error={{ message: errors.creditCard } as stripe.Error}
+                      onChange={({ error }) =>
```

Comment:
Did a little research code and, while this is definitely consistent with the rest of our credit card input invocations, it seems like `setFieldError` shouldn't really be used in `onChange`:

> setFieldError should not be used in change handlers because validation will not be aware of it during submission. The only real use case for setFieldError is inside of onSubmit where you want to manually set a field error. You should move this logic to validate instead.

(https://github.com/jaredpalmer/formik/issues/1278#issuecomment-461036355)

--

I think we should ship this PR as is, but just calling this out in case it's useful in the future.

Reply:
I don't think that comment applies to this specific callback because `CreditCardInput`'s `onChange` ([Strip Elements' own API](https://stripe.com/docs/stripe-js#react)) is fundamentally different from `input`'s `onChange` ([simulated native change event in React](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event)).

The real cause of the reported issue above is that both `setFieldValue` and `setFieldTouched` could invoke validation asynchronously and any value set by `setFieldError` could be unset by the validation that runs later on. Practically, both `setFieldValue` and `setFieldTouched` need to be called in `onChange` since it's really the only way to capture the input value. In that case, you shouldn't use `setFieldError`.

In the case of `CreditCardInput`, Stripe Elements encapsulates the validation logic and and we don't even have to capture the credit card number, etc, so the sole use of `setFieldError` in `CreditCardInput`'s `onChange` is appropriate.